grails-app/services/com/unifina/service/EthereumIntegrationKeyService.groovy

```diff
@@ -32,9 +33,26 @@ class EthereumIntegrationKeyService {
 
 	@PostConstruct
 	void init() {
-		String password = grailsApplication.config["streamr"]["encryption"]["password"]
+		initStringEncryptor()
+	}
+
+	private String getPassword() {
+		String password = Holders.getGrailsApplication().getConfig()["streamr"]["encryption"]["password"]
 		Assert.notNull(password, "streamr.encryption.password not set!")
-		encryptor = new StringEncryptor(password)
+		return password
+	}
+
+	//always instantiates
+	private initStringEncryptor(){
+		encryptor = new StringEncryptor(getPassword())
+	}
+
+	// instantiates only if encryptor doesn't exist
+	private StringEncryptor getStringEncryptor() {
+		if (encryptor == null) {
+			initStringEncryptor()
```

Comment:
I guess you could remove the one-liner function `initStringEncryptor` by simply replacing this line with:
```
encryptor = new StringEncryptor(getPassword())
```
and calling `getStringEncryptor()` on line 36 instead of `initStringEncryptor()`